src/libvirtApi/storageVolume.ts

```diff
@@ -49,38 +68,66 @@ export async function storageVolumeCreateAndAttach({
     cacheMode,
     busType,
     serial,
-}) {
+} : {
+    connectionName: ConnectionName,
+    poolName: string,
+    volumeName: string,
+    size: string,
+    format: string,
+    target: string,
+    vmId: string,
+    permanent: boolean,
+    hotplug: boolean,
+    cacheMode: string,
+    busType: string,
+    serial: string,
+}): Promise<void> {
```

Comment:
When it is like this it becomes a lot so I would split the typing to its own interface